src/js/components/AppVolumesListItemComponent.jsx

```diff
@@ -29,14 +29,14 @@ var AppVolumesListItemComponent = React.createClass({
 
     var params = {
       appId: encodeURIComponent(volume.appId),
-      volumeId: encodeURIComponent(volume.persistenceId)
+      volumeId: encodeURIComponent(volume.persistenceId || volume.external.name)
     };
 
     return (
       <tr>
         <td className={this.getHighlight("id")}>
           <Link to="volumeView" params={params}>
-            {volume.persistenceId}
+            {volume.persistenceId || volume.external.name}
```

Comment:
What if also `volume.external` is null? Let's be defensive and assign the volumeId to a variable before this.